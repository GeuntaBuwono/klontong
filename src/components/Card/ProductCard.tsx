import {rgbDataURL} from '@utils/rgbDataURL';
import Image from 'next/image';
import Link from 'next/link';

export const ProductCardSkeleton = () => (
	<div className="border border-blue-300 shadow rounded-md max-w-sm w-full mx-auto">
		<div className="animate-pulse flex flex-col">
			<div className="flex flex-1 items-center justify-center">
				<div className="bg-slate-700 h-60 w-full"></div>
			</div>
			<div className="flex-1 space-y-6 p-4">
				<div className="space-y-1">
					<div className="h-2 bg-slate-700 rounded col-span-2"></div>
					<div className="h-2 bg-slate-700 rounded col-span-1"></div>
				</div>

				<div className="space-y-1">
					<div className="h-2 bg-slate-700 rounded col-span-2"></div>
					<div className="h-2 bg-slate-700 rounded col-span-1"></div>
					<div className="h-2 bg-slate-700 rounded col-span-2"></div>
					<div className="h-2 bg-slate-700 rounded col-span-1"></div>
				</div>

				<div className="h-2 bg-slate-700 rounded"></div>
			</div>
		</div>
	</div>
);

type ProductCardProps = {
	onClickDelete: () => void;
	onClickEdit: () => void;
} & Omit<
	ProductType,
	| 'categoryName'
	| 'width'
	| 'weight'
	| 'sku'
	| 'length'
	| 'height'
	| 'CategoryId'
>;

export const ProductCard = ({
	id,
	image,
	description,
	name,
	harga,
	onClickEdit,
	onClickDelete,
}: ProductCardProps) => (
	<div className="bg-slate-100 text-white rounded-xl dark:bg-slate-800 md:max-w-md">
		<Link
			href={{
				pathname: '/dashboard/product/[id]',
				query: {id: String(id)},
			}}
		>
			{typeof image === 'string' && (
				<Image
					src={image}
					alt={description}
					blurDataURL={rgbDataURL(243, 243, 243)}
					width="512"
					height="250"
					placeholder="blur"
					style={{
						height: 'auto',
						maxHeight: '200px',
					}}
				/>
			)}
			<div className="px-6 py-4">
				<div className="mb-2">
					<p className="text-2xl font-semibold text-center line-clamp-1 capitalize">
						{name}
					</p>
					<p className="line-clamp-3">{description}</p>
				</div>
			</div>
			<div className="px-6 py-4 flex justify-between text-lg font-semibold">
				<span>Harga</span>
				<p>Rp.{harga}</p>
			</div>
		</Link>
		<div className="flex flex-1 justify-center">
			<button
				className="w-full py-2 bg-blue-600 text-white rounded-b-sm"
				onClick={onClickEdit}
			>
				<span>Edit</span>
			</button>
			<button
				className="w-full py-2 bg-red-600 text-white rounded-b-sm"
				onClick={onClickDelete}
			>
				<span>Remove</span>
			</button>
		</div>
	</div>
);
