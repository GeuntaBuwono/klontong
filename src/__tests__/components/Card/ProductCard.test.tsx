import {ProductCard, ProductCardSkeleton} from '@components/Card/ProductCard';
import {render, screen} from '@testing-library/react';

describe('ProductCardSkeleton', () => {
	const testElement = <ProductCardSkeleton />;
	test('snapshot', () => {
		expect(testElement).toMatchSnapshot();
	});

	test('render correctly', () => {
		render(testElement);
		const testInput = document.getElementsByClassName(
			'border border-blue-300 shadow rounded-md max-w-sm w-full mx-auto',
		);
		expect(testInput).toBeTruthy();
	});
});

describe('ProductCard', () => {
	const testElement = (
		<ProductCard
			description="testing description"
			onClickDelete={jest.fn()}
			name="testing name"
			harga={5000}
			id={1}
			image="https://loremflickr.com/640/480/abstract?random=1669555427495"
		/>
	);

	test('snapshot', () => {
		expect(testElement).toMatchSnapshot();
	});

	test('render correctly', () => {
		render(testElement);
		const testButton = screen.getByRole('button', {
			name: 'Remove Product',
		});

		const testImage = screen.getByRole('img', {
			name: 'testing description',
		});

		expect(testButton).toBeTruthy();
		expect(testImage).toBeTruthy();
	});
});
