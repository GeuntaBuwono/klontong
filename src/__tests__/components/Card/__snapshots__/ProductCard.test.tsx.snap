// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard snapshot 1`] = `
<ProductCard
  description="testing description"
  harga={5000}
  id={1}
  image="https://loremflickr.com/640/480/abstract?random=1669555427495"
  name="testing name"
  onClickDelete={[MockFunction]}
/>
`;

exports[`ProductCardSkeleton snapshot 1`] = `<ProductCardSkeleton />`;
